### ISSUE TITLE
Functions exported within immediately-invoked function expression lose their type checking

### ISSUE THREAD

**Issue by phistuck** [google/closure-compiler#911]
This code should produce warnings for the incorrect number of parameters and the type of the first parameter, but does not -
```
// ==ClosureCompiler==
// @compilation_level ADVANCED_OPTIMIZATIONS
// @output_file_name default.js
// ==/ClosureCompiler==

var globalInitialize;

(function () {
 /** @param {number} foo
     @param {number} bar
     @param {boolean} baz */
 function initialize(foo, bar, baz)
 {
  if (baz)
  {
   return foo + bar;
  }
  return 0;
 }

 globalInitialize = initialize;
}());

console.log(globalInitialize("string"));
```
